lib/client/indexer/market-clients/ammMarkets.ts

```diff
@@ -24,55 +26,72 @@ export class FutarchyIndexerAmmMarketsClient
   ): Promise<AmmMarket | undefined> {
     return this.rpcMarketsClient.fetchMarket(request);
   }
+
+  validateAddLiquidity(
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    baseAmount: number
+  ): LiquidityAddError | null {
+    return this.rpcMarketsClient.validateAddLiquidity(
+      ammMarket,
+      quoteAmount,
+      baseAmount
+    );
+  }
+
   async addLiquidity(
-    ammAddr: PublicKey,
-    quoteAmount?: number | undefined,
-    baseAmount?: number | undefined
-  ): Promise<string[]> {
-    return this.rpcMarketsClient.addLiquidity(ammAddr, quoteAmount, baseAmount);
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    baseAmount: number
+  ): Promise<string[] | LiquidityAddError> {
+    return this.rpcMarketsClient.addLiquidity(
+      ammMarket,
+      quoteAmount,
```

Comment:
Just as a reference, this needs to be calculated on the UI side OR / AND we can pass in a slippage param to the SDK such that we validate it in both areas... 

You quoteAmount amount is the `quoteAmount - (quoteAmount * slippage)`

You quoteAmount amount is the `baseAmount - (baseAmount * slippage)`

Reply:
These names should likely match what we have coming from the IX in the ts repo aas it's important to understanding the concepts.

Reply:
Names have been changed to match the amm client. LMK if you see something wrong here still.